packages/portalnetwork/src/wire/utp/PortalNetworkUtp/ContentRequest.ts

```diff
@@ -212,14 +229,41 @@ export class FindContentReadRequest extends ContentReadRequest {
         }
       }
       // If all packets have been received, return content
-      await this.returnContent([Uint8Array.from(this.socket.reader!.bytes)], [this.contentKey])
+      switch (this.version) {
+        case 0:
+          await this.returnContent([Uint8Array.from(this.socket.reader!.bytes)], [this.contentKey])
+          break
+        case 1: {
+          if (this.socket.reader!.contents.length > 0) {
+            const key = this.contentKey
+            const value = this.socket.reader!.contents.shift()!
```

Comment:
Why is this happening?  Are we assuming there's no more content to be received?

Reply:
Yes, this is a FindContent request, so 1 piece of content.  since we now know the length of that content via the prefix, we know when we have received the whole stream even before receiving the FIN